test/snippet/alignment/matrix/debug_matrix.cpp

```diff
@@ -0,0 +1,121 @@
+#include <iostream>
+
+#include <seqan3/alphabet/nucleotide/dna4.hpp>
+#include <seqan3/alignment/matrix/debug_matrix.hpp>
+#include <seqan3/io/stream/debug_stream.hpp>
+#include <seqan3/range/view/to_char.hpp>
+
+int main()
+{
+    using namespace seqan3;
+{
+    debug_stream << "//! [score_matrix::out]" << std::endl;
+//! [score_matrix]
+using namespace seqan3;
+using namespace seqan3::detail;
+
+std::vector<dna4> database = "AACCGGTT"_dna4;
+std::vector<dna4> query = "ACGT"_dna4;
+
+debug_matrix score_matrix
+{
+    std::vector
+    {
+        -0, -1, -2, -3, -4, -5, -6, -7, -8,
+        -1, -0, -1, -2, -3, -4, -5, -6, -7,
+        -2, -1, -1, -1, -2, -3, -4, -5, -6,
+        -3, -2, -2, -2, -2, -2, -3, -4, -5,
+        -4, -3, -3, -3, -3, -3, -3, -3, -4
+    },
+    5u,
+    9u
+};
+
+debug_stream << "database:\t" << (database | view::to_char) << std::endl;
+debug_stream << "query:\t\t" << (query | view::to_char) << std::endl;
+debug_stream << std::endl;
+
+debug_stream << "score_matrix: " << score_matrix.cols() << " columns and "
+          << score_matrix.rows() << " rows" << std::endl;
+
+// Print out the matrix.
+for(unsigned row = 0u; row < score_matrix.rows(); ++row)
+{
+    for(unsigned col = 0u; col < score_matrix.cols(); ++col)
```

Comment:
```suggestion
    for (unsigned col = 0u; col < score_matrix.cols(); ++col)
```

and also in a lot of for loops below